addons/stock/tests/test_stock_flow.py

```diff
@@ -2542,6 +2542,41 @@ def test_validate_backorder(self):
         backorder.button_validate()
         self.assertEqual(backorder.state, 'done')
 
+    def test_picking_mixed_tracking_with_backorder(self):
+        self.productB.tracking = 'lot'
+        picking = self.env['stock.picking'].create({
+            'location_id': self.supplier_location,
+            'location_dest_id': self.stock_location,
+            'picking_type_id': self.picking_type_in,
+            'company_id': self.env.company.id,
+        })
+        self.env['stock.move'].create({
+            'name': self.productA.name,
+            'product_id': self.productA.id,
+            'product_uom_qty': 1,
+            'product_uom': self.productA.uom_id.id,
+            'picking_id': picking.id,
+            'location_id': self.supplier_location,
+            'location_dest_id': self.stock_location,
+        })
+        self.env['stock.move'].create({
+            'name': self.productB.name,
+            'product_id': self.productB.id,
+            'product_uom_qty': 1,
+            'product_uom': self.productB.uom_id.id,
+            'picking_id': picking.id,
+            'location_id': self.supplier_location,
+            'location_dest_id': self.stock_location,
+        })
+        picking.action_confirm()
+
+        picking.move_ids[0].move_line_ids.write({'quantity': 1})
+        picking.move_ids[0].picked = True
```

Comment:
```suggestion
        no_tracking_move.picked = True
```
Since we're in reception, `quantity` will already be set to the `product_uom_qty`, no need to re-set it again. :slightly_smiling_face: